app/models/user.rb

```diff
@@ -0,0 +1,38 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  property :id, Serial
+  property :handle, String, required: true
+  property :email, String, required: true, unique: true
+  property :password_digest, Text
+
+  validates_presence_of :email
+  validates_presence_of :handle
+  validates_confirmation_of :password
+  validates_uniqueness_of :email
+  validates_uniqueness_of :handle
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.